lib/features/history/models/history_item.dart

```diff
@@ -17,6 +17,7 @@ abstract class HistoryItem extends Equatable {
 
 enum HistoryTypes {
   donation('WalletDonation'),
+  topUp('WalletTopup'),
   allowance('RecurringAllowance');
```

Comment:
I'll add one more comment that's related to this comment.

What I would like is there to be an "unknown/default" type. Right now we're really betting on the BE returning only these 3 types and that makes this part a bit fragile with any updates.

Reply:
well if it returns any other types they are not going top be added to the list because of this in the repo:
```
for (final donationMap in response) {
      if (donationMap['donationType'] == HistoryTypes.donation.value) {
        result.add(Donation.fromMap(donationMap));
      }
      if (donationMap['donationType'] == HistoryTypes.allowance.value ||
          donationMap['donationType'] == HistoryTypes.topUp.value) {
        result.add(Income.fromMap(donationMap));
      }
    }
```